src/clr-angular/forms/checkbox/checkbox.ts

```diff
@@ -4,17 +4,30 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Directive, ViewContainerRef } from '@angular/core';
+import { Directive, ViewContainerRef, Optional, Renderer2, ElementRef } from '@angular/core';
+import { NgControl } from '@angular/forms';
 
 import { WrappedFormControl } from '../common/wrapped-control';
-
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { NgControlService } from '../common/providers/ng-control.service';
+import { ControlClassService } from '../common/providers/control-class.service';
+import { MarkControlService } from '../common/providers/mark-control.service';
 import { ClrCheckboxContainer } from './checkbox-container';
 
 @Directive({ selector: '[clrCheckbox]' })
 export class ClrCheckboxNext extends WrappedFormControl<ClrCheckboxContainer> {
-  // Once again, several more elegant solutions were foiled by severity 3+ bugs on Angular that have been opened
-  // for 6 months to a year. So that's how we do it. Inheritance and ridiculous constructors. :-(
-  constructor(vcr: ViewContainerRef) {
-    super(ClrCheckboxContainer, vcr);
+  protected wrapperType = ClrCheckboxContainer;
+
+  constructor(
+    protected vcr: ViewContainerRef,
+    @Optional() ngControlService: NgControlService,
+    @Optional() ifErrorService: IfErrorService,
+    @Optional() control: NgControl,
+    @Optional() controlClassService: ControlClassService,
+    @Optional() markControlService: MarkControlService,
```

Comment:
Nitpick: when we inherit and have that many injections, I prefer by far just injecting `Injector`, passing that to the parent constructor, and letting it query all the services it needs.

The reason we don't do this for some instances is that you can't query for `ViewContainerRef`, `ElementRef` or similar DOM-related references through the injector, they have some kind of special status.